test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeDevice.cs

```diff
@@ -56,18 +57,32 @@ public class EdgeDevice
                     throw new InvalidOperationException($"Device '{device.Id}' exists, but is not an edge device");
```

Comment:
please also check IoT Edge capabilities in the other constructor.

Reply:
I could...but why? This is the only case where it might not be an edge device (the other case is where we create the device ourselves so we always create an edge device), so this is where we check.

Reply:
yes, you are right. that's a private constructor.